src/Runner.Common/JobServer.cs

```diff
@@ -45,8 +48,16 @@ public async Task ConnectAsync(VssConnection jobConnection)
                 }
                 catch (Exception ex) when (attemptCount > 0)
                 {
-                    Trace.Info($"Catch exception during connect. {attemptCount} attemp left.");
+                    Trace.Info($"Catch exception during connect. {attemptCount} attempts left.");
                     Trace.Error(ex);
+
+                    var configurationStore = HostContext.GetService<IConfigurationStore>();
+                    var runnerSettings = configurationStore.GetSettings();
+
+                    if (runnerSettings.IsHostedServer)
```

Comment:
I am worry about Mms is not populate this setting.

Reply:
I see. Do we have an alternative way to determine whether it's a hosted server? Should I just go with the checking the URI directly for a matched prefix?

Reply:
no worry, I just fix the service code.

Reply:
Ah, I missed this comment. I changed it to use `UrlUtil.IsHostedServer())`.